crates/ironrdp-pdu/src/utils.rs

```diff
@@ -23,6 +23,14 @@ pub fn combine_u64(lo: u32, hi: u32) -> u64 {
     u64::from_le_bytes(position_bytes)
 }
 
+pub fn to_utf8_bytes(value: &str, write_null_terminator: bool) -> Vec<u8> {
+    let mut bytes = value.as_bytes().to_vec();
+    if write_null_terminator {
+        bytes.push(0u8);
+    }
+    bytes
+}
```

Comment:
In my opinion, this should be more something like this, in order to opportunistically reuse the underlying buffer without extra allocation.
```suggestion
pub fn into_utf8_bytes(value: String, write_null_terminator: bool) -> Vec<u8> {
    let mut bytes = value.into_bytes();
    if write_null_terminator {
        bytes.push(0u8);
    }
    bytes
}
```
Though I’m not even sure how useful this helper will actually be in the future, and inlining the logic at the only callsite sounds reasonable to me as well in this case. When `write_null_terminator = false` I don’t think I would even bother using this function because `my_string.into_bytes()` / `my_string.as_bytes()` is already exactly that.

Reply:
b7a30f42cf0652e968b06e7033df01b8c1b6901a

Reply:
b7a30f42cf0652e968b06e7033df01b8c1b6901a